File: format-rmarkdown/src/main/java/digital/vasic/yole/format/rmarkdown/RmarkdownTextConverter.java
package digital.vasic.yole.format.rmarkdown;

import digital.vasic.yole.format.TextConverterBase;
import android.content.Context;
import java.io.File;

public class RmarkdownTextConverter extends TextConverterBase {
    @Override
    public boolean isFileOutOfThisFormat(File file) {
        // TODO: Implement file detection logic
        return false;
    }

    @Override
    public String convertMarkupToHtml(String markup, Context context, boolean lightMode, int lineNum, File file) {
        // TODO: Implement markup to HTML conversion
        return markup;
    }
}

File: app/src/main/kotlin/digital/vasic/yole/format/rmarkdown/RmarkdownTextConverter.kt
/*#######################################################
 *
 *   Maintained 2018-2025 by Gregor Santner <gsantner AT mailbox DOT org>
 *   License of this file: Apache 2.0
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 *########################################################*/
package digital.vasic.yole.format.rmarkdown

import android.content.Context
import digital.vasic.yole.format.markdown.MarkdownTextConverter
import java.io.File

/**
 * RMarkdown (R Markdown) text converter.
 *
 * RMarkdown is Markdown with embedded R code chunks.
 * This converter delegates to MarkdownTextConverter since RMarkdown
 * uses standard Markdown syntax with additional R code block support.
 */
class RmarkdownTextConverter : MarkdownTextConverter() {

    override fun convertMarkup(markup: String, context: Context, lightMode: Boolean, enableLineNumbers: Boolean, file: File): String {
        // RMarkdown is essentially Markdown, so delegate to parent
        return super.convertMarkup(markup, context, lightMode, enableLineNumbers, file)
    }

    override fun isFileOutOfThisFormat(file: File, name: String, ext: String): Boolean {
        return ext in EXT
    }

    companion object {
        private val EXT = setOf(".rmd", ".rmarkdown")
    }
}
